src/plugins/tables/tables.js

```diff
@@ -228,16 +228,49 @@ $document.on( "submit", ".wb-tables-filter", function( event ) {
 			return;
 		}
 
+		// Verifies if filtering the same column
+		if ( $column !== $prevCol || $prevCol === -1 ) {
+			$cachedVal = "";
+		}
+		$prevCol = $column;
+
 		// Filters based on input type
 		if ( $elm.is( "select" ) ) {
 			$value = $elm.find( "option:selected" ).val();
-		} else if ( $elm.is( ":checkbox" ) ) {
-
-			// Verifies if using same checkbox list
-			if ( $column !== $lastColumn || $lastColumn === -1 ) {
-				$cbVal = "";
+		} else if ( $elm.is( "input[type='date']" ) ) {
+			var $minDate, $maxDate, $fData;
+
+			// Retain minimum date (always the first date input)
+			if ( $cachedVal === "" ) {
+				$cachedVal = new Date( $elm.val() );
+				$cachedVal.setDate( $cachedVal.getDate() + 1 );
+				$cachedVal.setHours( 0, 0, 0, 0 );
 			}
-			$lastColumn = $column;
+			$minDate = $cachedVal;
+
+			// Maximum date is always the current selected date
+			$maxDate = new Date( $elm.val() );
+			$maxDate.setDate( $maxDate.getDate() + 1 );
+			$maxDate.setHours( 23, 59, 59, 999 );
+
+			// Generates a list of date strings (within the min and max date)
+			$fData = $datatable.column( $column ).data().filter( function( obj ) {
+				var $date = obj.replace( /[0-9]{2}\s[0-9]{2}\:/g, function( e ) {
+					return e.replace( /\s/g, "T" );
```

Comment:
We would need to documents the date format the author need to put into the data table in order to take advantage of this date filtering. Like:
* 2016-07-26 14:07:00
* 2016-07-26T14:07:00
* 2016-07-26

That documentation can be added in here : https://github.com/wet-boew/wet-boew/tree/master/site/pages/docs/ref/tables  or into the working example